runtime_scan/pkg/provider/gcp/scannerVm.go

```diff
@@ -0,0 +1,184 @@
+// Copyright © 2023 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"cloud.google.com/go/compute/apiv1/computepb"
+
+	"github.com/openclarity/vmclarity/runtime_scan/pkg/provider"
+	"github.com/openclarity/vmclarity/runtime_scan/pkg/provider/cloudinit"
+	"github.com/openclarity/vmclarity/shared/pkg/utils"
+)
+
+var (
+	VMCreateEstimateProvisionTime = 2 * time.Minute
+	VMDiskAttachEstimateTime      = 2 * time.Minute
+	VMDeleteEstimateTime          = 2 * time.Minute
+)
+
+func scannerVMNameFromJobConfig(config *provider.ScanJobConfig) string {
+	return fmt.Sprintf("vmclarity-scanner-%s", config.ScanResultID)
+}
+
+func (c *Client) ensureScannerVirtualMachine(ctx context.Context, config *provider.ScanJobConfig) (*computepb.Instance, error) {
+	vmName := scannerVMNameFromJobConfig(config)
+
+	instanceRes, err := c.instancesClient.Get(ctx, &computepb.GetInstanceRequest{
+		Instance: vmName,
+		Project:  c.gcpConfig.ProjectID,
+		Zone:     c.gcpConfig.ScannerZone,
+	})
+	if err == nil {
+		if *instanceRes.Status != InstanceStateRunning {
+			return instanceRes, provider.RetryableErrorf(VMCreateEstimateProvisionTime, "virtual machine is not ready yet. status: %v", *instanceRes.Status)
+		}
+
+		// Everything is good, the instance exists and running.
+		return instanceRes, nil
+	}
+
+	notFound, err := handleGcpRequestError(err, "getting scanner virtual machine: %v", vmName)
+	// ignore not found error as it is expected
```

Comment:
ignore **not found** error OR ignore **not not found** errors? :)

Reply:
ignore not found. let it sync :)

Reply:
my problem is that the comment is above `if !notFound {` so I'm a bit confused but it might just be me.